packages/fast-colors/src/color-palette.ts

```diff
@@ -158,10 +181,227 @@ export class ColorPalette {
+// If preserveInputColors is false the input colors are evenly distributed into
+// the output. Otherwise, the positions of the input colors are adjusted from
+// a perfectly even distribution in order to ensure that the exact color values
+// appearing in the input array also appear in the output array.
+// The larger targetSize is compared to input.length the smaller those adjustments
+// will be.
+export function rescale(
+    input: ColorRGBA64[],
+    targetSize: number,
+    preserveInputColors: boolean
+): ColorRGBA64[] {
+    if (input.length <= 1 || targetSize <= 1) {
+        throw new Error("The input array and targetSize must both be greater than 1");
+    }
+    if (preserveInputColors && targetSize <= input.length) {
+        throw new Error(
+            "If preserveInputColors is true then targetSize must be greater than the length of the input array"
+        );
+    }
+
+    const stops: ColorScaleStop[] = new Array(input.length);
+
+    if (preserveInputColors) {
+        for (let i: number = 0; i < input.length; i++) {
+            const p: number = i / (input.length - 1);
+            let bestFitValue: number = 2;
+            let bestFitIndex: number = 0;
+            for (let j: number = 0; j < targetSize; j++) {
+                const fitValue: number = Math.abs(j / (targetSize - 1) - p);
+                if (fitValue < bestFitValue) {
+                    bestFitValue = fitValue;
+                    bestFitIndex = j;
+                }
+                if (fitValue === 0) {
+                    break;
+                }
+            }
+            stops[i] = {
+                color: input[i],
+                position: bestFitIndex / (targetSize - 1),
+            };
+        }
+    } else {
+        for (let i: number = 0; i < stops.length; i++) {
+            stops[i] = { color: input[i], position: i / (input.length - 1) };
+        }
+    }
+
+    const scale: ColorScale = new ColorScale(stops);
+
+    const retVal: ColorRGBA64[] = new Array(targetSize);
+    for (let i: number = 0; i < targetSize; i++) {
+        retVal[i] = scale.getColor(i / (targetSize - 1));
+    }
+
+    return retVal;
+}
+
+// Takes an input array of colors and extrapolates them to a larger palette
+// The mapping first takes the input array and extrapolates between each color
+// so that they are separated by spacing-1 slots. Then it adds to either end enough
+// new colors to make up the desired targetSize. All output color slote between the
+// defined stops are interpolated.
+// EXAMPLE:
+// For an input array with length 5, a targetSize of 17 and spacing of 3 the output
+// would be:
+//  0: scaleColorLight
+//  1:
+//  2: input 0
+//  3:
+//  4:
+//  5: input 1
+//  6:
+//  7:
+//  8: input 2
+//  9:
+// 10:
+// 11: input 3
+// 12:
+// 13:
+// 14: input 4
+// 15:
+// 16: scaleColorDark
+export function centeredRescale(
+    input: ColorRGBA64[],
+    targetSize: number,

```

Comment:
Can we make this accept an object instead? That will allow us to add to this over time if we want without making a breaking change. It also makes it easier to use if I want to only use a dark scale color.